app/main/views/services.py

```diff
@@ -432,6 +433,7 @@ def update_section_submission(service_id, section_id):
     else:
         update_data.update(uploaded_documents)
 
+    if not errors and has_changes_to_save(section, draft, update_data):
```

Comment:
I didn't want to nest the API request even more and I like the way documents_errors check has two clear outcomes now.